pkg/api/v1beta2/dynakube/validation/namespace_selector.go

```diff
@@ -22,9 +23,16 @@ func conflictingNamespaceSelector(ctx context.Context, dv *Validator, dk *dynaku
 		return ""
 	}
 
-	dkMapper := mapper.NewDynakubeMapper(ctx, nil, dv.apiReader, dk.Namespace, dk)
+	dst := &v1beta3.DynaKube{}
 
-	_, err := dkMapper.MatchingNamespaces()
+	err := dk.ConvertTo(dst)
```

Comment:
IMO, this is fine, as we are comparing a new/updated dynakube againts all others present, and because of the stored version being `v1beta3` it is the version that all other dynakubes are (even if some had to be converted), so it is a nicer thing to convert 1 dynakube (the one that is being validated) to the stored version, than converting every other dynakube to the old version.
- saves us a bunch of network traffic (each conversion is a call to the webhook, if requested through the API)

We can either:
- Remove this check(and similar) from older versions, with the reason that "if validation does comparisons with other CRs it should only be done for the stored version of that CR"
  - I think doing a quick conversion here is cheaper and clearer than "letting the API" deal with it
- Just update the unit-test, as `TestConflictingNamespaceSelector/invalid_dynakube_specs` will not call conversion logic on it's on, the mocks are good, but not that good 😅

Reply:
fixed in 965bd51c